Delivery API: Add section about deeply nested JSON (#5325)
umbraco/UmbracoDocs: 12/umbraco-cms/reference/content-delivery-api/README.md

* [Tailor the API's response for custom property editors](custom-property-editors-support.md)
* [Extend the API with custom selecting, filtering, and sorting options](extension-api-for-querying.md)

## Handling deeply nested JSON output

> .NET imposes a limit on the depth of object nesting within rendered JSON. This is done in an effort to detect cyclic references (read more [here](https://learn.microsoft.com/en-us/dotnet/api/system.text.json.jsonserializeroptions.maxdepth)). If the limit is exceeded, .NET will throw a `JsonException`.

Reviewer: ```suggestion
.NET imposes a limit on the depth of object nesting within rendered JSON. This is done in an effort to detect cyclic references. Learn more about it in [the official .NET API docs](https://learn.microsoft.com/en-us/dotnet/api/system.text.json.jsonserializeroptions.maxdepth)).

If the limit is exceeded, .NET will throw a `JsonException`.
```

In some cases the content models might be so deeply nested that the Delivery API produces JSON that exceeds this limit. If this happens, the `JsonException` will be logged and shown in the Umbraco log (in the _Settings_ section).

To handle this we have to change the limit. Since the Delivery API has its own JSON configuration, we can do so without affecting the rest of our site.

> First we have to add these `using` statements to `Startup.cs`:.

Reviewer: ```suggestion
First, we have to add these `using` statements to `Startup.cs`:
```

* [Tailor the API's response for custom property editors](custom-property-editors-support.md)
* [Extend the API with custom selecting, filtering, and sorting options](extension-api-for-querying.md)

## Handling deeply nested JSON output

> .NET imposes a limit on the depth of object nesting within rendered JSON. This is done in an effort to detect cyclic references. Learn more about it in [the official .NET API docs](https://learn.microsoft.com/en-us/dotnet/api/system.text.json.jsonserializeroptions.maxdepth)).

Reviewer 2: seems there is an extra ")" at the end 😄 
```suggestion
.NET imposes a limit on the depth of object nesting within rendered JSON. This is done in an effort to detect cyclic references. Learn more about it in [the official .NET API docs](https://learn.microsoft.com/en-us/dotnet/api/system.text.json.jsonserializeroptions.maxdepth).
```
